packages/checkout/widgets-lib/src/widgets/wallet/components/WalletAddress/WalletAddress.tsx

```diff
@@ -1,13 +1,56 @@
-import { Box, Body, Heading } from '@biom3/react';
+import { MenuItem, ButtCon } from '@biom3/react';
 import { Web3Provider } from '@ethersproject/providers';
 import { useEffect, useState } from 'react';
+import { WalletProviderName } from '@imtbl/checkout-sdk';
+import { getWalletLogoByName } from 'lib/logoUtils';
+import { useTranslation } from 'react-i18next';
+import { abbreviateWalletAddress } from 'lib/utils';
 import {
-  walletAddressContainerStyle, walletAddressLayoutStyle, walletAddressTextLayoutStyle, truncatedTextStyle,
-} from './WalletAddressStyles';
-import { CopyButton } from './CopyButton';
+  UserJourney,
+  useAnalytics,
+} from '../../../../context/analytics-provider/SegmentAnalyticsProvider';
 
-export function WalletAddress({ provider }: { provider?: Web3Provider }) {
+const isCopiedStyle = {
+  background: 'base.color.status.success.bright',
+  fill: 'base.color.status.success.bright',
+};
+
+export function WalletAddress({
+  provider,
+  showL1Warning,
+  setShowL1Warning,
+}: {
+  provider?: Web3Provider;
+  showL1Warning: boolean;
+  setShowL1Warning: (show: boolean) => void;
+}) {
   const [walletAddress, setWalletAddress] = useState<string>('');
+  const [isCopied, setIsCopied] = useState(false);
+
+  const { t } = useTranslation();
+
+  const { track } = useAnalytics();
+
+  const walletProviderName = () => {
+    if ((provider?.provider as any)?.isMetaMask) {
+      return WalletProviderName.METAMASK;
+    }
+    if ((provider?.provider as any)?.isPassport) {
```

Comment:
there is an existing util here
https://github.com/immutable/ts-immutable-sdk/blob/9142ba1180244f38e5b6a56ca72d2f6623c6e699/packages/checkout/widgets-lib/src/lib/provider/utils.ts#L4-L4